device/coordinate_manager.h

```diff
@@ -45,6 +46,17 @@ class CoordinateManager {
 
```

Comment:
Just talked offline with @tt-vjovanovic 
Let's consider changing get_coordinate_manager to pass either ARCH + harvesting, or just a yaml soc descriptor. We wouldn't pass worker_grid_size nor worker_grid_size in both options.